src/notebook/notebook/actions.ts

```diff
@@ -195,18 +193,12 @@ namespace NotebookActions {
     for (let cell of toDelete) {
       cells.remove(cell);
     }
-
-    // Add a new code cell if all cells were deleted.
-    if (!model.cells.length) {
-      let cell = model.factory.createCodeCell();
-      model.cells.add(cell);
-    }
+    // The model will add a new code cell if there are no
+    // remaining cells.
     model.cells.endCompoundOperation();
 
-    // Activate the previous cell.
-    if (index === -1) {
-      index = 0;
-    }
+    // Select the cell *after* the last selected.
```

Comment:
Can you add a comment that clamping the index on setting it ensures that if the last cell was deleted, the new last cell is selected? That part is a bit magical, and it makes sure we are clear for the future on what the behavior should be.